kafka-rest/src/main/java/io/confluent/kafkarest/resources/v3/AclsResource.java

```diff
@@ -212,7 +212,7 @@ public void createAcl(
   public void deleteAcls(
       @Suspended AsyncResponse asyncResponse,
       @PathParam("clusterId") String clusterId,
-      @QueryParam("resource_type") @DefaultValue("any") ResourceType resourceType,
+      @QueryParam("resource_type") @DefaultValue("unknown") ResourceType resourceType,
```

Comment:
Question 1: Should we also change `patternType`, `operation` and `permission` to `unknown`, for consistency?

Question 2: Is this backwards compatible? I don't think so. We might be forced to just update the documentation to explain it better. :(

Reply:
In the Java API, you have to explicitly set all fields. There are no defaults. The defaults here are mostly for convenience, I believe. For the read operation, that's fine, I guess. But for this one here, we probably want to force something to be specified.

Now, another option to just forcing `resourceType`, or forcing everything, is to just force something. i.e.: all the defaults stay the same, but you have to set at least one field, anyone will do. That eliminates the empty == all case.

As for backwards compatibility, you are correct. I think it's fine to change the defaults if they aren't documented.

Reply:
> The defaults here are mostly for convenience, I believe.

What I meant was mostly related to this:
_For some reason, requests with partially filled query params are not currently hitting the "Delete ACL" endpoint - it should be investigated whether that's due to a testing problem, Jersey bug, or something else._
My original desire was to add to the "Delete ACL" integration tests a test that does the following:
1. Sets up a couple ACLs.
2. Tries to delete all of them with empty query params and fails.
3. Tries to delete all of them by just specifying `?resource_type=all` and omitting the rest of the query params.

For some reason, while 2. worked as expected (and still is part of the test that we have in this change), 3. did not hit the "Delete ACL" endpoint. This was pretty surprising to me, but I did not spend too much time investigating it, as I still want to focus on other things this Sprint. Only when I specified all other params did I hit the endpoint. At that point I decided to just reuse the common search criteria.

> Now, another option to just forcing resourceType, or forcing everything, is to just force something. i.e.: all the defaults stay the same, but you have to set at least one field, anyone will do.

I am not sure if this is better, but more importantly I am not sure how can we implement it. I.e. how do we differentiate between all the parameters having the default values because of an empty request vs because the client has set exactly a single parameter to its default value?